pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -53,13 +57,15 @@ var (
 
 type IngressReconciler struct {
 	client        client.Client
+	clientset     kubernetes.Interface
 	scheme        *runtime.Scheme
 	ingressConfig *v1beta1.IngressConfig
 }
 
-func NewIngressReconciler(client client.Client, scheme *runtime.Scheme, ingressConfig *v1beta1.IngressConfig) *IngressReconciler {
+func NewIngressReconciler(client client.Client, clientset kubernetes.Interface, scheme *runtime.Scheme, ingressConfig *v1beta1.IngressConfig) *IngressReconciler {
 	return &IngressReconciler{
 		client:        client,
+		clientset:     clientset,
```

Comment:
may be nice to add a note as to why we want to use both client and clientset, just reading the code it can be confusing to see, but (correct me if im wrong) its because client/clientset have access to different functions/resources

Reply:
I think we only need to use controller runtime client as we do not use lower level clientset across kserve project.